### PR TITLE
[FLOC-1965] Release/flocker 0.4.1dev1

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1965.

Build results at http://build.clusterhq.com/boxes-flocker?branch=release%2Fflocker-0.4.1dev1 shows 1 failed test outside of "Expected failures" - `flocker.node.agents.functional.test_cinder.CinderBlockDeviceAPIInterfaceTests.test_attach_volume_validate_size` on the `flocker/functional/redhat-openstack/centos-7/storage-driver` builder. I reopened https://clusterhq.atlassian.net/browse/FLOC-1914 for this.

## Update

Despite the problems below I think that this should be merged, and perhaps a follow-up issue should be created. I think that the first part of the review (checking links) would be a good time to review the PR for automation of that by using it: https://clusterhq.atlassian.net/browse/FLOC-1701.

#### Problems

**TLS issue with `vagrant up` in Getting Started**

The getting started guide does not work because of https://clusterhq.atlassian.net/browse/FLOC-1932.
https://clusterhq.atlassian.net/browse/FLOC-1307 should probably be done before the next release to avoid breaking old docs.

This means that most of the guide has not been tested.

**Ubuntu / Fedora install issue in Getting Started**

I tried to run Ubuntu instructions from Getting Started guide.
This was in a container started with `docker run -i -t ubuntu /bin/bash`.
They failed with:

```
Successfully installed pip
Cleaning up...
/flocker-tutorial/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/flocker-tutorial/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/flocker-tutorial/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Command "/flocker-tutorial/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-build-i7S868/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-2kcr89-record/install-record.txt --single-version-externally-managed --compile --install-headers /flocker-tutorial/include/site/python2.7/cffi" failed with error code 1 in /tmp/pip-build-i7S868/cffi
```

Similarly on Fedora:

```
/flocker-tutorial/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/flocker-tutorial/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Command "/flocker-tutorial/bin/python2 -c "import setuptools, tokenize;__file__='/tmp/pip-build-4MYt7O/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-66uYDc-record/install-record.txt --single-version-externally-managed --compile --install-headers /flocker-tutorial/include/site/python2.7/cffi" failed with error code 1 in /tmp/pip-build-4MYt7O/cffi
```

I then tried with Ubuntu on Vagrant:

```
$ vagrant init ubuntu/trusty64
$ vagrant up

vagrant@vagrant-ubuntu-trusty-64:~$
```

and this had the same problem.

I assume that a follow up issue is necessary, but I'm not sure that this was the right way to test this. Also, because https://clusterhq.atlassian.net/browse/FLOC-1742 seemed to be going smoothly maybe the issue for Ubuntu has been fixed in that.

If this is a suitable way to test, don't do that part of the review and I'll create follow up issues.
If this is not a suitable way to test, please tell me what is and I will document it to save a future release engineer this hassle.

### CODE REVIEW

File: docs/advanced/whatsnew.rst [hunk 1]
```diff
@@ -6,10 +6,17 @@ What's New
...
-====
+Next Release
+============
 
 * ``flocker-deploy`` now takes an extra argument, the hostname of the control service.
+* It is now necessary to specify a dataset backend for each agent node.
+  See :ref:`post-installation-configuration`.
+* Flocker-initiated communication is secured with TLS.
+  See :ref:`authentication`.
+* Docker restart policies are adhered to.
```
Comment (wallrj) [inline docs/advanced/whatsnew.rst]: :question: This was just a bug fix. Probably not worth noting in the What's New section.

**Comment by wallrj** [comment]
> flocker.node.agents.functional.test_cinder.CinderBlockDeviceAPIInterfaceTests.test_attach_volume_validate_size on the flocker/functional/redhat-openstack/centos-7/storage-driver builder. I reopened https://clusterhq.atlassian.net/browse/FLOC-1914 for this.

That builder is in the "configured" expected failures list in lastpass.
@tomprince recently renamed some of the builders though.
FLOC-1946 is where we'll fix that error. 
FLOC-1914 should be closed again.

Otherwise, this all looks good to me.

Please proceed.

**Comment by adamtheturtle** [comment]
I think that the What's New can and will be revisited before 1.0 so I'll leave the optional comment for the sake of expedience. Thanks for the review.

**Comment by adamtheturtle** [comment]
I created https://clusterhq.atlassian.net/browse/FLOC-1978 as a follow-up to the issues mentioned in the description.

**Comment by adamtheturtle** [comment]
The problem with the unrelated commits has hit my other PRs too. I think when recovering from yesterday's accidental force push perhaps. With the others I just merge master in, but can't really do that here because it is a release.

**Comment by adamtheturtle** [comment]
Actually, as the tag was made first, it should be fine to merge master into this branch.

**Comment by wallrj** [comment]
> I tried to run Ubuntu instructions from Getting Started guide.
This was in a container started with docker run -i -t ubuntu /bin/bash.
They failed with:
> ...
> I assume that a follow up issue is necessary, but I'm not sure that this was the right way to test this. Also, because https://clusterhq.atlassian.net/browse/FLOC-1742 seemed to be going smoothly maybe the issue for Ubuntu has been fixed in that.

The problem is that we require libssl-dev(el) and libffi-dev(el) since introducing `flocker-ca` in https://github.com/ClusterHQ/flocker/pull/1295

I added some notes to:
 * https://clusterhq.atlassian.net/browse/FLOC-1978

@jongiddy 's new instructions (https://clusterhq.atlassian.net/browse/FLOC-1742) won't encounter this because the cffi "bindings" (?) have been precompiled during omnibus packaging.

**Comment by wallrj** [comment]
Thanks @adamtheturtle 

 * I tested the devel redirects manually (they work) and 
 * tried using your new tool, it has some issues (https://github.com/ClusterHQ/flocker/pull/1449#issuecomment-106015066)
 * Made some notes on why the CLI can't be installed with pip.

But as agreed, we're going to ignore those errors for this release and fix them for the next one.

I'll merge.

**Pull request merged**